crates/router/src/core/admin.rs

```diff
@@ -2027,6 +2027,30 @@ impl MerchantConnectorAccountUpdateBridge for api_models::admin::MerchantConnect
 
         pm_auth_config_validation.validate_pm_auth_config().await?;
 
+        let merchant_recipient_data = if let Some(data) = &self.additional_merchant_data {
+            Some(
+                process_open_banking_connectors(
+                    state,
+                    merchant_account.get_id(),
+                    &auth,
+                    &self.connector_type,
+                    &connector_enum,
+                    types::AdditionalMerchantData::foreign_from(data.clone()),
+                )
+                .await?,
+            )
+        } else {
+            None
+        }
```

Comment:
Since `additional_merchant_data` could contain other info (types) too including open banking types, its better to perform match on `additional_merchant_data` here in parent function rather than passing the entire type to `process_open_banking_connectors` and matching there. This helps in easily extending `additional_merchant_data` with other types. 
Not a blocker though

Reply:
Right, we'll add the required casing as more variants are added.